libraries/typescript/Microsoft.Bot.Builder.Adapters.WeChat/src/weChatAdapter.ts

```diff
@@ -0,0 +1,373 @@
+     * @param wechatRequest Request message entity from wechat.
+     * @param logic A function handler that will be called to perform the bots logic after the the adapters middleware has been run.
+     * @param passiveResponse Marked the message whether it needs passive reply or not.
+     * @returns Response message entity.
+     */
+    public async ProcessWeChatRequest(wechatRequest: IRequestMessageBase, logic: (context: TurnContext) => Promise<any>, passiveResponse: boolean): Promise<any> {
+        const activity = await this.weChatMessageMapper.ToConnectorMessage(wechatRequest);
+        const context = new TurnContext(this, activity as Activity);
+        const responses = new Map<string, Array<Activity>>();
+        context.turnState.set(this.TurnResponseKey, responses);
+        await this.runMiddleware(context, logic);
+        const key = `${ activity.conversation.id }:${ activity.id }`;
+        let activities: any;
+        activities = responses.has(key) ? responses.get(key) : new Array<Activity>();
+        const response = await this.ProcessBotResponse(activities, wechatRequest.FromUserName, passiveResponse);
+        return response;
+    }
+
+    /**
+     * Does not support by WeChat.
+     */
+    public async deleteActivity(context: TurnContext, reference: Partial<ConversationReference>): Promise<void> {
+        throw new Error('WeChat does not support deleting activities.');
+    }
+
+    /**
+     * Does not support by WeChat.
+     */
+    public async updateActivity(context: TurnContext, activity: Partial<Activity>): Promise<void> {
+        throw new Error('WeChat does not support updating activities.');
+    }
+
+    /**
+     * Sends a set of outgoing activities to WeChat.
+     * @param context Context for the current turn of conversation with the user.
+     * @param activities List of activities to send.
+     * @returns Response activities
+     */
+    public async sendActivities(context: TurnContext, activities: Activity[]): Promise<ResourceResponse[]> {
+        const resourceResponses: ResourceResponse[] = [];
+
+        for (let i = 0; i < activities.length; i++) {
+            const activity: Activity = activities[i];
+
+            switch (activity.type) {
+                case ActivityTypes.Message:
+                case ActivityTypes.EndOfConversation:
+                    const conversation = activity.conversation;
+                    const key = `${ conversation.id }:${ activity.replyToId }`;
+                    const responses = context.turnState.get(this.TurnResponseKey);
+                    if (responses.has(key)) {
+                        responses.get(key).push(activity);
+                    } else {
+                        responses.set(key, [activity]);
+                    }
+                    break;
+                default:
+                    break;
+            }
+            const resourceResponse: ResourceResponse = {
+                id: activity.id || ''
```

Comment:
>activity.id || '' [](start = 20, length = 17)

This is expected to be the ID returned by the service. Since WeChat doesn't support update or delete you can just set this to undefined versus an empty string.